bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java

```diff
@@ -65,7 +65,7 @@
     private final Logger logger = LoggerFactory.getLogger(HomeAssistantDiscovery.class);
     protected final Map<String, Set<HaID>> componentsPerThingID = new TreeMap<>();
     protected final Map<String, ThingUID> thingIDPerTopic = new TreeMap<>();
-    protected final Map<String, DiscoveryResult> results = new TreeMap<>();
+    protected final ConcurrentHashMap<String, DiscoveryResult> results = new ConcurrentHashMap<>();
```

Comment:
Keep the `Map` in the declaration. But another suggestion:

Why don‘t you create an inner class 

```
public static class HADiscoveryResult {
    public final Set<HaId> components = new HashSet<>();
    public @Nullable DiscoveryResult discoveryResult;
}
```

and use one `Map<ThingUID, HADiscoveryResult>` instead of the two sets you have now? Makes it easier to be thread safe late when clearing the `Map`.

Reply:
Hi @J-N-K!

I have changed the ```results``` declaration type, but I would prefer not to change the rest of the implementation, as it works fine. Do you think it's necessary?